Python/Product/Analysis/AnalysisValueSetExtensions.cs

```diff
@@ -259,6 +274,60 @@ public static class AnalysisValueSetExtensions {
             return res;
         }
 
+        /// <summary>
+        /// Gets the returned value for a yield from.
+        /// </summary>
+        public static IAnalysisSet GetReturnForYieldFrom(this IAnalysisSet self, Node node, AnalysisUnit unit) {
+            var res = AnalysisSet.Empty;
+            foreach (var ns in self) {
+                res = res.Union(ns.GetReturnForYieldFrom(node, unit));
+            }
+
+            return res;
+        }
+
+        public static IAnalysisSet Resolve(this IAnalysisSet self, AnalysisUnit unit) => Resolve(self, unit, ResolutionContext.Complete, out _);
+
+        internal static IAnalysisSet Resolve(this IAnalysisSet self, AnalysisUnit unit, ResolutionContext context, out bool changed) {
+            // The vast majority of the time, no values are resolved
+            // So we want to quickly validate and get out without allocating
+            // or changing anything.
+
+            if (!context.Push()) {
+                changed = false;
+                return self;
+            }
+            try {
+                List<AnalysisValue> removed = null;
+                IAnalysisSet added = null;
+                foreach (var ns in self) {
```

Comment:
Why variable name is `ns`?

Reply:
Legacy reasons mainly. This type used to be `Namespace` instead of `AnalysisValue` and `NamespaceSet` instead of `AnalysisSet`. I did a huge refactor a few years ago, but there was nothing gained by changing all the loop variables.